app/helpers/ticketing.py

```diff
@@ -303,9 +303,9 @@ def initiate_order_payment(form, paid_via=None):
         if order:
             user = DataGetter.get_or_create_user_by_email(email, form)
             order.user_id = user.id
-            if not(order.user.user_detail.firstname or order.user.user_detail.lastname):
-                 order.user.user_detail.firstname = form['firstname']
-                 order.user.user_detail.lastname = form['lastname']
+            if not (order.user.user_detail.firstname or order.user.user_detail.lastname):
```

Comment:
No. in some cases, there maybe a firstname but no lastname. We don't want the name to get overwritten in those cases. 

Overwrite only if both firstname and lastname is not present in `user_detail`